crypto/threads_pthread.c

```diff
@@ -67,6 +67,13 @@ __tsan_mutex_post_lock((x), 0, 0)
 #  define USE_RWLOCK
 # endif
 
+/*
+ * The AARCH64 atomics seem to be broken
+ */
+# ifdef __aarch64__
+#  define USE_ATOMIC_FALLBACKS
+# endif
+
```

Comment:
It would be nice if we had a comment in the code that pointed to what we think the break in atomics is here. 

FWIW, I just ran across this:
https://lists.infradead.org/pipermail/linux-arm-kernel/2014-February/229588.html

In trying to better understand the situation.  It seems to allude to the same problem you have been describing.  The kernel solved it by re-implementing their atomics for arm as full barriers

Reply:
Yeah, this somehow matches to my theory how the acquire barrier might be in the middle of the atomic_add:
https://github.com/openssl/openssl/pull/26919#issuecomment-2691275845
but here we have a single atomic instruction, and the magnitude of the probability of that effect that I had
observed on my MACOS M1 experiments makes me wonder if there is a barrier at all.

Reply:
Some context here is that the Linux kernel uses a different memory model from the standard C11 one.
https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2020/p0124r7.html